core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -415,31 +323,41 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
 
     setName(threadName)
 
+    private def toBaseConsumerRecord(record: ConsumerRecord[Array[Byte], Array[Byte]]): BaseConsumerRecord =
+      BaseConsumerRecord(record.topic,
+        record.partition,
+        record.offset,
+        record.timestamp,
+        record.timestampType,
+        record.key,
+        record.value,
+        record.headers)
+
     override def run() {
       info("Starting mirror maker thread " + threadName)
       try {
-        mirrorMakerConsumer.init()
+        consumerWrapper.init()
 
-        // We need the two while loop to make sure when old consumer is used, even there is no message we
-        // still commit offset. When new consumer is used, this is handled by poll(timeout).
+        // FIXME Is it safe to remove one of the loops?
```

Comment:
Originally we need two while loops because `mirrorMakerConsumer.receive ` will block infinitely for old consumer if there is no data. I think it should be safe to just remote the second while loop.

Reply:
Thanks for the explanation. I'll keep the existing code (will just update the comment), but will file a JIRA for simplifying MirrorMaker.